OpenSim/Actuators/VandenBogert2011Muscle.cpp

```diff
@@ -0,0 +1,1234 @@
+    double fiberExp = (fiberLengthNorm - 1.0) / fl_width;   // [dimensionless]
+    double F1 = exp(-pow(fiberExp, 2));        // Gaussian force-length curve
+
+    double dF1_dfiberLengthNorm = 0;
+    if (returnJacobians) {
+        dF1_dfiberLengthNorm = -2.0 * fiberExp * F1 / fl_width;
+        double dF1_dprojFibLenNorm =
+                dF1_dfiberLengthNorm * dfiberLength_dprojFibLen;
+    }
+
+
+
+    //-------- F2 is the dimensionless force-velocity relationship -------//
+    double F2;
+    double dF2_dfiberVelocityNorm;
+    double dF2_dactiv;
+    double dF2_dprojFibVelNorm;
+    double dF2_dprojFibLenNorm;
+    double df_dmuscleLength;
+
+    // Chow/Darling Vel-Activation Relationship //TODO:  Add full reference
+    //double lambda = 0.5025 + 0.5341*activ;
+    //double  dlambda_da = 0.5341;
+    double lambda = 1;   //Turn it off for now as it seems to cause an issue
+                        // with negative force large concentric vel
+    double dlambda_da =0;
+
+    if (fiberVelocityNorm < 0) {
+        //Hill's equation for concentric contraction
+        // F2 = (V_{max} + V_{fiber}) / (V_{max} - V_{fiber}/a_{Hill})
+
+        double hillDenom = (lambda*vMaxNorm - fiberVelocityNorm/fv_AHill);
+        F2 = (lambda*vMaxNorm + fiberVelocityNorm) / hillDenom;
+
+        if (returnJacobians) {
+            dF2_dfiberVelocityNorm  = (1.0 + F2 / fv_AHill) / hillDenom;
+            dF2_dactiv = - dlambda_da * vMaxNorm * fiberVelocityNorm *
+                    (1.0 + 1.0/fv_AHill) / pow(hillDenom,2);
+        }
+    }
+    else {
+        //Katz Curve for eccentric contraction
+        // c is Katz Constant
+        double c3 = vMaxNorm * fv_AHill * (fv_maxMultiplier - 1.0) /
+                (fv_AHill + 1.0); // parameter in the eccentric f-v equation
+        double c = lambda*c3;
+        //F2 = (g_{max} * V_{fiber} + c) / (V_{fiber} + c)
+        double katzDenom = (fiberVelocityNorm  + c);
+        F2 = (fv_maxMultiplier * fiberVelocityNorm  + c) / katzDenom ;
+        if (returnJacobians) {
+            dF2_dfiberVelocityNorm = (fv_maxMultiplier - F2) / katzDenom ;
+            dF2_dactiv = dlambda_da * c3 * fiberVelocityNorm *
+                    (1-fv_maxMultiplier) / pow(katzDenom,2);
+        }
+    }
+    if (returnJacobians){
+        dF2_dprojFibVelNorm =
+                dF2_dfiberVelocityNorm * dfiberVelocityNorm_dprojFibVelNorm;
+        dF2_dprojFibLenNorm =
+                dF2_dfiberVelocityNorm * dfiberVelocityNorm_dprojFibLenNorm;
+    }
+
+
+    //------F3 is the dimensionless fiber (PEE) force (in units of Fmax)------//
+
+
+
+    double dF3_dprojFibLenNorm;
+
+    // stiffness of the linear term is 1 N/m, convert to Fmax/Lceopt units
+    double kPEE = 1.0 / maxIsoForce * optFiberLength;
+    // elongation of fiber (PEE), relative to Lceopt
+    double kPEE2Norm = 1 / pow(fl_width, 2);  //Fiber (PEE) quadratic stiffness,
+    //          so Fpee = Fmax when Lce = Lce*(1+W)
```

Comment:
I also expect the value of `Lce` that causes `Fpee = Fmax` to depend on `fiberSlackLengthNorm`.